com.microsoft.mrtk.spatialmanipulation/ObjectManipulator/ObjectManipulator.cs

```diff
@@ -570,106 +639,154 @@ public override void ProcessInteractable(XRInteractionUpdateOrder.UpdatePhase up
             else
             {
                 // There is a Rigidbody. Potential different paths for near vs far manipulation
-                if (IsGrabSelected && !useForcesForNearManipulation)
+                if (!useForces)
                 {
-                    rigidBody.MovePosition(targetPose.Position);
-                    rigidBody.MoveRotation(targetPose.Rotation);
+                    rigidBody.MovePosition(targetTransform.Position);
+                    rigidBody.MoveRotation(targetTransform.Rotation);
                 }
-                else
-                {
-                    // We are using forces
-                    rigidBody.velocity = ((1f - Mathf.Pow(moveLerpTime, Time.deltaTime)) / Time.deltaTime) * (targetPose.Position - HostTransform.position);
 
-                    var relativeRotation = targetPose.Rotation * Quaternion.Inverse(HostTransform.rotation);
-                    relativeRotation.ToAngleAxis(out float angle, out Vector3 axis);
+                HostTransform.localScale = targetTransform.Scale;
+            }
+        }
+
+        /// <summary>
+        /// In case a Rigidbody gets the targetTransform applied using physical forcees, this function is called within the
+        /// FixedUpdate() routine with physics-conforming time stepping.
+        /// </summary>
+        void ApplyForcesToRigidbody()
+        {
+            var referenceFrameVelocity = referenceFrameHasLastPos
+                ? (referenceFrameTransform.position - referenceFrameLastPos) / Time.fixedDeltaTime
+                : Vector3.zero;
+
+            referenceFrameLastPos = referenceFrameTransform.position;
+            referenceFrameHasLastPos = true;
 
-                    if (axis.IsValidVector())
+            // implement critically dampened spring force, scaled to mass-independent frequency
+            float omega = Mathf.PI / springForceSoftness;  // angular frequency, sqrt(k/m)
+
+            Vector3 distance = HostTransform.position - targetTransform.Position;
+
+            // when player is moving, we need to anticipate where the targetTransform is going to be one time step from now
+            distance -= referenceFrameVelocity * Time.fixedDeltaTime;
+
+            // Apply damping - mathematically, we need e^(-2 * omega * dt)
+            // To compensate for the finite time step, this is split in two equal factors,
+            // one applied before, the other after the spring force
+            // equivalent with applying damping as well as spring force continuously
+            float halfDampingFactor = Mathf.Exp(-springForceDamping * omega * Time.fixedDeltaTime);
+
+            var velocity = rigidBody.velocity;
+
+            velocity -= referenceFrameVelocity;  // change to the player's frame of reference before damping
+
+            var oldVelocity = velocity;
+
+            velocity *= halfDampingFactor;  // 1/2 damping
+            velocity -= distance * omega * omega * Time.fixedDeltaTime; // spring force
```

Comment:
divide by dt instead of multiply to get from distance to a velocity (?)

Reply:
I just had a long discussion with Oliver Strunk and it turns out that I'm really using not just a different formalism but fundamentally a different algorithm than that commonly used in our own codebase. The dt factor here is correct.